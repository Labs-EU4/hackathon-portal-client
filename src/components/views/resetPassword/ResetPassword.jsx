import React from "react";
import { Formik, Form, ErrorMessage } from "formik";
import * as Yup from "yup";
import { useDispatch } from "react-redux";
import { useHistory } from "react-router-dom";

import {BodyContainer} from "../../../assets/styles/atoms/BodyContainer";
import { RowBody } from "../../../assets/styles/atoms/RowBody";
import { ErrorSpan } from "../../../assets/styles/atoms/Span";
import {Container} from "../../../assets/styles/atoms/Container";
import { H1 } from "../../../assets/styles/atoms/HeadingStyling";
import { Paragraph } from "../../../assets/styles/atoms/Paragraph";
import Button from "../../atoms/Button";
import Input from "../../atoms/Input";
import HeroImage from "../../atoms/HeroImage";
import image from "../../../assets/images/Password.png";
import { forgotPassword } from "../../../store/user/actions";

const ResetPassword = () => {
  const dispatch = useDispatch();
  const history = useHistory();

  const handleSubmit = values => {
    const { email } = values;
    dispatch(forgotPassword(email, history));
  };

  const schema = Yup.object().shape({
    email: Yup.string()
      .email("Please use a valid email address.")
      .required("Email address is required.")
  });

  return (
    <BodyContainer justify="center">
      <HeroImage src={image} alt="Reset passowrd" />
      <Container>
        <H1>Reset the password</H1>
        <Paragraph>
          Enter your email address so we can reset your password and send a link to your inbox.
        </Paragraph>
        <Formik
          initialValues={{ email: "" }}
          onSubmit={handleSubmit}
          validationSchema={schema}
        >
          {({ errors, touched }) => (
            <Form>
              <Input display="wide" id="email" type="email" name="email" placeholder="Email address"/>
              {errors.name && touched.name ? (
                <div>{errors.name}</div>
              ) : null}
              <ErrorSpan>
                <ErrorMessage name="email" />
              </ErrorSpan>
              <RowBody>
                <Button size="wide" type="submit" color="blue">
                  Reset Password
                </Button>
              </RowBody>
            </Form>
          )}
        </Formik>
      </Container>
    </BodyContainer>
  );
};

export default ResetPassword;