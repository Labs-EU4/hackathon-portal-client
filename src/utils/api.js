import Axios from "axios";
import { toast } from "react-toastify";
import { resetUser } from "../store/user/actions";

const baseURL = process.env.REACT_APP_API_URL;

// if process.env.CONTEXT = "production" {
//   Point to heroku API
// }
// if process.env.BRANCH = "develop" {
  // point to Heroku staging API
// }
// if process.env.CONTEXT = "deploy-preview" {
//   point to deploy-preview HEROKU API (the separate one)
// }



export const axiosWithAuth = token =>
  Axios.create({
    baseURL,
    headers: {
      Authorization: token
    }
  });

export const axios = Axios.create({ baseURL });

export const selectToken = state => state.currentUser.token;

export const showError = message => {
  toast.error(message);
};

export const showSuccess = message => {
  toast.success(message);
};

export function* handleError({ response }, put, history) {
  const { message, statusCode } = response.data;
  if (statusCode === 404) {
    history.push("/not-found");
  }
  if (message === "jwt expired") {
    yield put(resetUser());
  } else showError(`⚠️ ${message}`);
}
