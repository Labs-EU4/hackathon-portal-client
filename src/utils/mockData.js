export const initialState = {
  events: {
    data: [
      {
        id: 1,
        event_title: "EuroHack",
        event_description: "the best event hack hack hack macbook",
        creator_id: 1,
        start_date: "10/09/11",
        end_date: "08/07/13",
        location: "Europe",
        guidelines: "cool hekhkdsjfhsdkjf skdjfhskdjfhkjsdfh skjdfhksjdfh",
        participation_type: "Team",
        category_id: 1,
        requirements: [
          {
            video_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        ProfileImage: [
          {
            image_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        rubrics: [
          "Presentation",
          "Product Fit",
          "Functionality",
          "Innovation",
          "Product Design",
          "Extensibility"
        ]
      },
      {
        id: 2,
        event_title: "World",
        event_description: "number one hacker event in the world",
        creator_id: 1,
        start_date: "07/09/11",
        end_date: "08/07/13",
        location: "Asia",
        guidelines: "come well prepared,no cheating,no fighting",
        participation_type: "Team",
        category_id: 2
      }
    ],
    categories: [
      { id: 2, category_name: "Summer Hackton" },
      { id: 3, category_name: "World Hackton" },
      { id: 3, category_name: "Asia Hackton" },
      { id: 3, category_name: "Gamer's Hackton" },
      { id: 3, category_name: "Advanced Hackton" }
    ]
  },

  currentUser: {
    token: 347534574536,
    userId: 1,
    fullname: "Jake Tom",
    email: "8omemail@google.com",
    username: "Jake22",
    bio: "I like coding ",
    image: ""
  },
  submissions: [
    {
      id: 1,
      project_title: "Mock Project",
      participant_or_team_name: "Mock Team",
      event_id: 7,
      video_url: "",
      git_url: "https://www.google.com/",
      project_writeups: "No mock write up",
      submitted_by: 14,
      created_at: "2020-03-02",
      updated_at: "2020-03-02",
      average_rating: 0,
      acted_judges: 0,
      number_of_judges: 0
    }
  ],

  teams: [
    { id: 1, team_name: "The A team", team_lead: 2 },
    { id: 2, team_name: "Hacking central", team_lead: 3 },
    { id: 3, team_name: "blackhats", team_lead: 4 },
    { id: 4, team_name: "Spies", team_lead: 1 }
  ],
  participant_team_members: [
    {
      team_member: 3,
      team_id: 1
    },
    {
      team_member: 4,
      team_id: 1
    },
    {
      team_member: 5,
      team_id: 1
    }
  ]
};
