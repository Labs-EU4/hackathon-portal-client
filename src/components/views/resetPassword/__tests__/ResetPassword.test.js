import React from "react";
import { Router } from "react-router";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import "@testing-library/jest-dom/extend-expect";
import ResetPassword from "../ResetPassword";
import { initialState } from "../../../../utils/mockData";

const history = createMemoryHistory();

let component;
let mockStore;
let store;

beforeEach(() => {
  mockStore = configureStore();
  store = mockStore(initialState);
  component = render(
    <Router history={history}>
      <Provider store={store}>
        <ResetPassword />
      </Provider>
    </Router>
  );
});

describe("Asserts that text nodes on ResetPassword.js,render properly", () => {
  it("ResetPassword.js component renders properly", () => {
    expect(component).toMatchSnapshot();
  });
  it("The text node for the <H1> Reset the password, renders properly ", () => {
    expect(component.queryByText(/Reset the password/i)).toBeInTheDocument();
  });
  it("The text node for the <paragraph> enter email address, renders properly ", () => {
    expect(
      component.queryByText(
        /Enter your email address so we can reset your password and send a link to your inbox./i
      )
    ).toBeInTheDocument();
  });
  //   it("The text node for the <h3> Participant Teams text node, renders properly", () => {
  //     expect(component.queryByText(/Participant Teams/i)).toBeInTheDocument();
  //   });
  //   it("The text node for the <h4> You are creating a team for text node, renders properly", () => {
  //     expect(
  //       component.queryByText(/You are creating a team for/i)
  //     ).toBeInTheDocument();
  //   });
  //   it("The text node for the team being created for the selected event 'EuroHack' <span>, renders properly", () => {
  //     expect(component.queryByText(/EuroHack/i)).toBeInTheDocument();
  //   });
  //   it("The text node for the Team Name <label> input, renders properly", () => {
  //     expect(component.queryByText(/Team Name/i)).toBeInTheDocument();
  //   });
  //   it("The role for the button used to submit the form to create a team renders properly", () => {
  //     expect(component.getByRole(/button/i)).toBeInTheDocument();
  //   });
  //   it("The text node for the submit button <button>, renders properly", () => {
  //     expect(component.getByText(/submit/i)).toBeInTheDocument();
  //   });
});
