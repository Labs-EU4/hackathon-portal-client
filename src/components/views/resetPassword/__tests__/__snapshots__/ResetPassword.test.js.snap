// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asserts that text nodes on ResetPassword.js,render properly ResetPassword.js component renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdVaJa ezLspW"
      >
        <img
          alt="Reset passowrd"
          class="sc-kAzzGY eAsvzO"
          src="Password.png"
        />
        <div
          class="sc-htoDjs cMiEQS"
        >
          <h1
            class="sc-dnqmqq eiIxdn"
          >
            Reset the password
          </h1>
          <p
            class="sc-VigVT fMEWJn"
          >
            Enter your email address so we can reset your password and send a link to your inbox.
          </p>
          <form
            action="#"
          >
            <input
              class="sc-cSHVUG kikqQz"
              display="wide"
              id="email"
              name="email"
              placeholder="Email address"
              type="email"
              value=""
            />
            <span
              class="sc-bxivhb sc-gzVnrw dSolTc"
            />
            <div
              class="sc-htpNat jNuDmB"
            >
              <button
                class="sc-jTzLTM rPycD"
                color="blue"
                type="submit"
              >
                Reset Password
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdVaJa ezLspW"
    >
      <img
        alt="Reset passowrd"
        class="sc-kAzzGY eAsvzO"
        src="Password.png"
      />
      <div
        class="sc-htoDjs cMiEQS"
      >
        <h1
          class="sc-dnqmqq eiIxdn"
        >
          Reset the password
        </h1>
        <p
          class="sc-VigVT fMEWJn"
        >
          Enter your email address so we can reset your password and send a link to your inbox.
        </p>
        <form
          action="#"
        >
          <input
            class="sc-cSHVUG kikqQz"
            display="wide"
            id="email"
            name="email"
            placeholder="Email address"
            type="email"
            value=""
          />
          <span
            class="sc-bxivhb sc-gzVnrw dSolTc"
          />
          <div
            class="sc-htpNat jNuDmB"
          >
            <button
              class="sc-jTzLTM rPycD"
              color="blue"
              type="submit"
            >
              Reset Password
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
